tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py

```diff
@@ -213,6 +213,17 @@ def find_subchannel_with_state(self, channel: _ChannelzChannel,
             f'Not found a {_ChannelzChannelState.Name(state)} '
             f'subchannel for channel_id {channel.ref.channel_id}')
 
+    def find_subchannels_with_state(self, state: _ChannelzChannelState,
+                                    **kwargs) -> List[_ChannelzSubchannel]:
+        channels = self.channelz.find_channels_for_target(
+            self.server_target, **kwargs)
+        subchannels = self.channelz.list_channel_subchannels(
+            next(channels), **kwargs)
```

Comment:
Do we want to check if there are multiple channels connecting to the same target?

Reply:
There could be channels shared the same target but with different channel options. Could happen in future for a testing framework.